### PR TITLE
added a RelativeIndex() func to helmhome to return a relative path to…

### PR DESCRIPTION
**Pull request by Stelminator**
… the index files. Also updated repo_add.go to use this new fun

part 2 of #3327: this makes `repo add` write a path relative to the cache directory into `repositories.yaml`

### CODE REVIEW

File: pkg/helm/helmpath/helmhome_unix_test.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ func TestHelmHome(t *testing.T) {
 	isEq(t, hh.LocalRepository(), "/r/repository/local")
 	isEq(t, hh.Cache(), "/r/repository/cache")
 	isEq(t, hh.CacheIndex("t"), "/r/repository/cache/t-index.yaml")
+	isEq(t, hh.RelativeIndex("t"), "t-index.yaml")
```
Comment (bacongobbler) [inline pkg/helm/helmpath/helmhome_unix_test.go]: hmm... Wouldn't we want the relative path be equal to `repository/cache/t-index.yaml`, as in relative to the home directory's path?

Comment (bacongobbler) [reply pkg/helm/helmpath/helmhome_unix_test.go]: It'd make sense when relative to repositories.yaml, but not the home directory.

It's all relative :)

**Comment by bacongobbler** [comment]
Perhaps a better function call would be

`home.RelativeJoin(home.RepositoryFile(), home.CacheIndex(name))`

Which would return a relative path from file A (in this case, repositories.yaml) to file B (index.yaml).

In the current design, `home.RelativeIndex()` will always output the name of the file, which I don't think is entirely useful as a function...

How does that sound?

File: pkg/helm/helmpath/helmhome_unix_test.go (hunk 1 above)
Comment (Stelminator) [reply pkg/helm/helmpath/helmhome_unix_test.go]: it has to be relative to `hh.Cache()`, due to https://github.com/kubernetes/helm/blob/8cf72adf521311225274fa835a2b04060eeb781a/pkg/repo/chartrepo.go#L136-L138

**Comment by Stelminator** [comment]
@bacongobbler @thomastaylor312 please take a look at this updated PR, I think just changing the name of the function to `CacheRelativeIndex` makes it read a lot easier.  I also used it to build `CacheIndex` to put it in context.

**Review by thomastaylor312 — APPROVED** [review]
This is looking good to me. I'll let @bacongobbler do the final review and merge since he started this

**Comment by Stelminator** [comment]
after discussion with @bacongobbler on slack, closing this in favor of https://github.com/kubernetes/helm/pull/3613 to avoid adding a function to `pkg/helm/helmpath/helmhome.go`

**Pull request closed without merging**